flycheck.el

```diff
@@ -8704,12 +8725,40 @@ rustc command.  See URL `https://www.rust-lang.org'."
   :error-parser flycheck-parse-rust
   :error-explainer flycheck-rust-error-explainer
   :modes rust-mode
-  :predicate (lambda ()
-               ;; Since we build the entire project with cargo rustc we require
-               ;; that the buffer is saved.  And of course, we also need a Cargo
-               ;; file :)
-               (and (flycheck-buffer-saved-p)
-                    (locate-dominating-file (buffer-file-name) "Cargo.toml"))))
+  ;; Since we build the entire project with cargo rustc we require
+  ;; that the buffer is saved.
+  :predicate flycheck-buffer-saved-p
+  :enabled flycheck-rust-find-cargo-root
+  :verify (lambda(_) (list (flycheck-rust--verify-cargo-toml))))
+
+(flycheck-define-checker rust-cargo-clippy
+  "A Rust syntax checker and linter using clippy.
+
+This syntax checker needs the cargo clippy subcommand. See URL `https://github.com/Manishearth/rust-clippy'"
```

Comment:
Please wrap this line; it's too long and goes beyond our 80 character limit.  Just press `M-q` to automatically wrap the string :)